spec/integration/type/file_spec.rb

```diff
@@ -268,6 +275,17 @@ def get_aces_for_path_by_sid(path, sid)
           expect(get_mode(path) & 07777).to eq(0666)
         end
 
+        context "file is in protected windows directory", :if => Puppet.features.microsoft_windows? do
+          after { FileUtils.rm(path_protected) }
+
+          it "should set and get the correct mode for files inside protected windows folders" do
+            catalog.add_resource described_class.new(:path => path_protected, :ensure => :file, :mode => '0644')
+            catalog.apply
+  
+            expect(get_mode(path_protected) & 07777).to eq(0644)
```

Comment:
Ruby defaults to `0644` for a newly created files:

```
irb(main):002:0> (File.stat('C:\Users\josh\puppet\Gemfile').mode & 0777).to_s(8)
=> "644"
```

So it's possible this test will pass without the puppet code working. I'd suggest breaking this into 2 tests 1) that sets the mode to 0640, and verifies puppet can change the mode to 0644, and 2) setting the mode to 0644, and verifying the mode stays at 0644 and the resource's status is unchanged, like https://github.com/puppetlabs/puppet/blob/dbc130168be2d030caf70176e958af96d87f82f1/spec/integration/type/file_spec.rb#L1342